### ISSUE TITLE
Cannot create indexes on Collection Columns (Set, List)

### ISSUE THREAD

**Issue by gkgupta** [doanduyhai/Achilles#218]
Adding @Index on top of Collection Type columns gives following error on server startup:

```java
Caused by: info.archinnov.achilles.exception.AchillesBeanMappingException: Property 'numbers' of entity 'com.example.User' cannot be indexed because the type 'java.util.Set' is not supported
	at info.archinnov.achilles.internal.validation.Validator.validateBeanMappingTrue(Validator.java:126)
	at info.archinnov.achilles.internal.metadata.parsing.validator.PropertyParsingValidator.validateIndexIfSet(PropertyParsingValidator.java:118)
	at info.archinnov.achilles.internal.metadata.parsing.PropertyParser.parse(PropertyParser.java:178)
	at info.archinnov.achilles.internal.metadata.parsing.EntityParser.parseEntity(EntityParser.java:65)
	at info.archinnov.achilles.internal.metadata.discovery.AchillesBootstrapper.buildMetaDatas(AchillesBootstrapper.java:55)
	at info.archinnov.achilles.persistence.PersistenceManagerFactory.parseEntities(PersistenceManagerFactory.java:125)
	at info.archinnov.achilles.persistence.PersistenceManagerFactory.bootstrap(PersistenceManagerFactory.java:95)
	at info.archinnov.achilles.persistence.PersistenceManagerFactory$PersistenceManagerFactoryBuilder.build(PersistenceManagerFactory.java:269)
	at info.archinnov.achilles.integration.spring.PersistenceManagerFactoryBean.initialize(PersistenceManagerFactoryBean.java:135)
	at info.archinnov.achilles.integration.spring.PersistenceManagerFactoryBean.createInstance(PersistenceManagerFactoryBean.java:286)
	at info.archinnov.achilles.integration.spring.PersistenceManagerFactoryBean.createInstance(PersistenceManagerFactoryBean.java:58)
	at org.springframework.beans.factory.config.AbstractFactoryBean.afterPropertiesSet(AbstractFactoryBean.java:134)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1625)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1562)
```

**Comment by doanduyhai** [comment]
It's normal, it's not supported. Why ? Because **Achilles** provides very **limited** support for secondary index. Why ? Because they're tricky to use and do not scale very well. Documentation here: https://github.com/doanduyhai/Achilles/wiki/Entity-Mapping#secondary-index

**Comment by doanduyhai** [comment]
As a work-around, you can always create the index manually and use TypedQuery to query them

**Comment by gkgupta** [comment]
Of course these indexes can be created manually. But having @Index and supporting it for raw types and not for collections is not really fair. As the secondary index performance issue is there for both collection and raw types.
I can do it manually. But I was relying on Achilles generated DDL for validating my schema integrity. And in case there is a need for explicitly making the developer choose. There could be a configuration parameter where indexes for collections can be enabled (disabled by default) along side indexes for raw types.

**Comment by doanduyhai** [comment]
Validating secondary index schema for collection is quite hard, it's not impossible but very tricky. You can make a pull request to improve Achilles for support of those indices if you wish

**Comment by doanduyhai** [comment]
Fixed in Achilles 4.0.0
